platform/pulp/platform/models/task.py

```diff
@@ -0,0 +1,298 @@
+    :type name: models.TextField
+
+    :cvar last_heartbeat: A timestamp of this worker's last heartbeat
+    :type last_heartbeat: models.DateTimeField
+    """
+    name = models.TextField(db_index=True, unique=True)
+    last_heartbeat = models.DateTimeField()
+
+
+class TaskLock(Model):
+    """
+    Locking mechanism for services that utilize active/passive fail-over
+
+    Fields:
+
+    :cvar name: The name of the item that has the lock
+    :type name: models.TextField
+
+    :cvar timestamp: The time the lock was acquired
+    :type timestamp: models.DateTimeField
+
+    :cvar lock: The name of the lock acquired
+    :type lock: models.TextField
+
+    """
+    CELERY_BEAT = 'CeleryBeat'
+    RESOURCE_MANAGER = 'ResourceManager'
+    LOCK_STRINGS = (
+        (CELERY_BEAT, 'Celery Beat Lock'),
+        (RESOURCE_MANAGER, 'Resource Manager Lock')
+    )
+
+    name = models.TextField(db_index=True, unique=True)
+    timestamp = models.DateTimeField(auto_now_add=True)
+    lock = models.TextField(unique=True, null=False, choices=LOCK_STRINGS)
+
+
+class Task(Model):
+    """
+    Represents a task
+
+    Fields:
+
+    :cvar task_type: The type of task this is
+    :type task_type: models.TextField
+
+    :cvar task_group: The group this task belongs to
+    :type task_group: models.UUIDField
+
+    :cvar state: The state of the task
+    :type state: models.TextField
+
+    :cvar start_time: The time the task started executing
+    :type start_time: models.DateTimeField
+
+    :cvar finish_time: The time the task finished executing
+    :type finish_time: models.DateTimeField
+
+    :cvar error: Serialized collection of errors that might have occurred while task was running
+    :type error: models.TextField
+
+    :cvar result: Serialized return value of the task
+    :type result: models.TextField
+
+    Relations:
+
+    :cvar parent: Task that spawned this task (if any)
+    :type parent: models.ForeignKey
+
+    :cvar worker: The worker that this task is in
+    :type worker: models.ForeignKey
+    """
+
+    WAITING = 'waiting'
+    SKIPPED = 'skipped'
+    ACCEPTED = 'accepted'
+    RUNNING = 'running'
+    SUSPENDED = 'suspended'
+    FINISHED = 'finished'
+    ERROR = 'error'
+    CANCELED = 'canceled'
+    STATES = (
+        (WAITING, 'Waiting'),
+        (SKIPPED, 'Skipped'),
+        (ACCEPTED, 'Accepted'),
+        (RUNNING, 'Running'),
+        (SUSPENDED, 'Suspended'),
+        (FINISHED, 'Finished'),
+        (ERROR, 'Error'),
+        (CANCELED, 'Canceled')
+    )
+    task_type = models.TextField()
+    task_group = models.UUIDField()
+    state = models.TextField(choices=STATES)
+
+    start_time = models.DateTimeField()
+    finish_time = models.DateTimeField()
+
+    _error = models.TextField(null=True, db_column='error')
+    _result = models.TextField(null=True, db_column='result')
+
+    parent = models.ForeignKey("Task", null=True)
+    worker = models.ForeignKey(Worker)
+
+    @property
+    def result(self):
```

Comment:
I suggested the same. I think the note on the PR says he's still investigating options. Thanks for pointing out that ``JSONField``!